corehq/toggles/__init__.py

```diff
@@ -637,6 +637,13 @@ def _ensure_valid_randomness(randomness):
     [NAMESPACE_DOMAIN],
 )
 
+USE_CUSTOM_EXTERNAL_ID_CASE_PROPERTY = StaticToggle(
```

Comment:
The idea of using the toggle to make the user aware of an edge-case doesn't seem to fit purpose of a feature-flag.

Is it possible to go with option-1 but to make aware of the user about the edge-case, display a warning to the the user to select explicitly the system's `external_id` or the user defined `external_id` if they use `external_id` for configuring the rules.

Reply:
Agreed. As a "make aware" strategy I do agree it's not the best to simply chuck it behind a FF, but I thought this behaviour is strange enough to justify a feature flag. Although if you feel like going the "display warning" route I'm open to make some changes.

Reply:
@sravfeyn Since this PR is rather urgent (needs to be deployed this week), I'm wondering if we should not go ahead with the current solution, then I'll just follow up with another PR? (I'm a little busy so can't immediately attend to this).

Reply:
@Charl1996 Is there a ticket to look at this again in order to remove it from the feature flag and do it as suggested above?

Reply:
If there's a ticket, I see no problem with merging this if it's time sensitive

Reply:
I created a ticket [here](https://dimagi-dev.atlassian.net/browse/SC-2431) which speaks to
1. adding the warning display in case future projects define a case property named `external_id`
2. removing the feature flag if deemed unnecessary